types/appconfig.pp

```diff
@@ -0,0 +1,24 @@
+type Puppetwebhook::Config = Struct[{
+  protected                 => Boolean,
+  user                      => String,
+  pass                      => String,
+  client_cfg                => Stdlib::Absolutepath,
+  client_timeout            => String,
+  use_mco_ruby              => Boolean,
+  use_mcollective           => Boolean,
+  discovery_timeout         => String,
+  chatops                   => Boolean,
+  Optional[chatops_service] => Enum['slack', 'rocketchat'],
+  Optional[chatops_channel] => Pattern[/\A#.*/],
+  Optional[chatops_user]    => String,
+  Optional[chatops_url]     => Variant[Stdlib::HTTPUrl, Stdlib::HTTPSUrl],
+  default_branch            => String,
+  ignore_environments       => Array,
```

Comment:
Do we know the datatypes within the arrays?

Reply:
Probably `String[1]` Other than that, probably not since they are open-ended and dependent on what users name their environments